### PR TITLE
fix: Progress throws warning `findDOMNode is deprecated in StrictMode`

### LINKED ISSUE #42110: 搭配试用Progress和Tooltip组件，控制台报错：findDOMNode is deprecated in StrictMode.
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-7j08s7?file=/index.js)

### Steps to reproduce

Tooltip组件和Progress组件搭配试用

### What is expected?

解决报错问题

### What is actually happening?

报错信息如下：

```
Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of DomWrapper which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node
    at div
```

| Environment | Info |
| --- | --- |
| antd | 5.4.6 |
| React | 18.2.0 |
| System | Win11 |
| Browser | Edge |

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix https://github.com/ant-design/ant-design/issues/42110

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix Progress throws warning `findDOMNode is deprecated in StrictMode`      |
| 🇨🇳 Chinese |     修复Progress抛出的警告 "findDOMNode is deprecated in StrictMode"     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 33029d7</samp>

Added ref forwarding and a new test case for the `Progress` component. This improves the component's compatibility with React strict mode and other components that use refs, such as `Tooltip`.

### <samp>🤖 Generated by Copilot at 33029d7</samp>

*  Wrap `Progress` component in `React.forwardRef` and pass `ref` prop to `div` element to support ref forwarding ([link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL8-R10),[link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL58-R58),[link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddR175),[link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL191-R192))
* Import `Tooltip` component from `antd` and add test case for `Progress` component wrapped in `Tooltip` component to check for no strict mode warning ([link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-c045f3b22eef7546211dddb79e6b4195b1e0a3dd6af6ae640c55579f9eeb5116R1),[link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-c045f3b22eef7546211dddb79e6b4195b1e0a3dd6af6ae640c55579f9eeb5116R363-R380))
* Move `ProgressTypes` type from `progress.tsx` to `index.test.tsx` to avoid circular dependency issue ([link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-c045f3b22eef7546211dddb79e6b4195b1e0a3dd6af6ae640c55579f9eeb5116L8-R14),[link](https://github.com/ant-design/ant-design/pull/42241/files?diff=unified&w=0#diff-faba3b9fef75b1f2b3c50badd4cb81e9d9a3ac66376bcf8e3059ed5cc1c8c1ddL8-R10))

### CODE REVIEW

File: components/progress/__tests__/index.test.tsx [hunk 1]
```diff
@@ -360,4 +360,22 @@ describe('Progress', () => {
...
+      <Tooltip title="当前已使用60%">
+        <Progress percent={60} type="circle" />
+      </Tooltip>,
+    );
+    rerender(
+      <Tooltip title="当前已使用60%">
+        <Progress percent={60} type="circle" />
+      </Tooltip>,
+    );
+    expect(errSpy).not.toHaveBeenCalledWith(
```
Comment (li-jia-nan) [inline components/progress/__tests__/index.test.tsx:376]: 如果是 not 的话，后面填一堆报错信息就没啥意义了吧，不如直接 `expect(errSpy).not.toHaveBeenCalled();`

**Review by li-jia-nan — APPROVED** [review]

File: components/progress/__tests__/index.test.tsx [hunk 2]
```diff
@@ -360,4 +360,22 @@ describe('Progress', () => {
...
+      'Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of DomWrapper which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node',
+    );
```
Comment (afc163) [inline components/progress/__tests__/index.test.tsx:378]: 这个 warning 过于完整了，如果 react 改个字符就对不上了。

Comment (afc163) [inline components/progress/__tests__/index.test.tsx:378]:
```
.not.toHaveBeenCalledWith(expect.stringContaining('findDOMNode is deprecated in StrictMode'))
```

Comment (afc163) [reply components/progress/__tests__/index.test.tsx:378]:
```
.not.toHaveBeenCalledWith(expect.stringContaining('findDOMNode is deprecated in StrictMode'))
```

**Pull request merged**